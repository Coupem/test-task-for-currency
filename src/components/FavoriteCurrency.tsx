// eslint-disable-next-line @typescript-eslint/no-use-before-define
import React from 'react';
import PropTypes from 'prop-types';

import { ListGroup } from 'react-bootstrap';
import {
  FavoriteCurrencyWrapper,
  NoFavoriteCurrency,
  TitleFavoriteCurrency,
  ListGroupContainer,
} from '../styled';
import { FavoriteCurrencyProp } from '../types';

const FavoriteCurrency = ({
  favoriteCurrencies,
}: FavoriteCurrencyProp): JSX.Element => (
  <FavoriteCurrencyWrapper>
    <TitleFavoriteCurrency>Favorite Currency:</TitleFavoriteCurrency>

    {Object.values(favoriteCurrencies).length === 0 && (
      <NoFavoriteCurrency>NO CURRENCY</NoFavoriteCurrency>
    )}

    {Object.values(favoriteCurrencies).map((breakpoint) => (
      <ListGroupContainer key={breakpoint.ccy}>
        <ListGroup.Item>Currency: {breakpoint.ccy}</ListGroup.Item>
        <ListGroup.Item>Buy: {breakpoint.buy}</ListGroup.Item>
        <ListGroup.Item>Sale: {breakpoint.sale}</ListGroup.Item>
      </ListGroupContainer>
    ))}
  </FavoriteCurrencyWrapper>
);

FavoriteCurrency.propTypes = {
  favoriteCurrencies: PropTypes.shape({
    ccy: PropTypes.string,
    buy: PropTypes.string,
    sale: PropTypes.string,
  }),
};

export default FavoriteCurrency;
