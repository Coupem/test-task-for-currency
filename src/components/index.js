export { default as InputValuesGroup } from './Converter/InputValuesGroup';
export { default as ConvertCurrency } from './Converter/ConvertCurrency';
export { default as FavoriteCurrency } from './Course/FavoriteCurrency';
export { default as Navigation } from './Navigation/Navigation';
export { default as Header } from './Header/Header';
export { default as Footer } from './Footer/Footer';
