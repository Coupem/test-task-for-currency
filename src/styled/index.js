export * from './MainLayout/MainContentWrapper';
export * from './MainLayout/InteractionContainerWrapper';
export * from './Footer/FooterComponent';
export * from './Header/HeaderComponent';
export * from './Navigation/NavButton';
export * from './Navigation/NavigationComponent';
export * from './Course/FavoriteCurrencyWrapper';
export * from './Course/ButtonBest';
export * from './Course/NoFavoriteCurrency';
export * from './Course/TitleCourseComponent';
export * from './Course/TitleFavoriteCurrency';
export * from './Course/ValidCourseContainer';
export * from './Converter/ConverterContainer';
export * from './Converter/TitleConverterContainer';
export * from './Course/ListTableCurrency';
export * from './MainLayout/MainLayoutContainer';
export * from './Converter/InputGroupContainer';
export * from './Converter/TableCurrencyContainer';
export * from './Course/ListGroupContainer';
export * from './Converter/InputContainer';
export * from './Converter/SelectContainer';
export * from './Converter/ErrorContainer';
export * from './Converter/OptionContainer';
