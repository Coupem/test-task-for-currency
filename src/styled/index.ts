export * from './Footer/FooterComponent';
export * from './Header/HeaderComponent';
export { NavButton, NavigationComponent } from './Navigation/navigationStyled';
export {
  MainContentWrapper,
  InteractionContainerWrapper,
  MainLayoutContainer,
} from './MainLayout/mainLayout';
export {
  FavoriteCurrencyWrapper,
  ButtonBest,
  NoFavoriteCurrency,
  TitleCourseComponent,
  TitleFavoriteCurrency,
  ValidCourseContainer,
  ListGroupContainer,
  ListTableCurrency,
} from './Course/courseStyled';
export {
  TitleConverterContainer,
  InputGroupContainer,
  TableCurrencyContainer,
  SelectContainer,
  InputContainer,
  ErrorContainer,
  ConverterContainer,
  OptionContainer,
} from './Converter/converterStyled';
