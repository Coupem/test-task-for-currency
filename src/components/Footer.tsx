import React from 'react';
import { FooterComponent } from '../styled';

const Footer = (): JSX.Element => (
  <FooterComponent>Â© 2021 My. Built My Theme.</FooterComponent>
);

Footer.propTypes = {};

export default Footer;
