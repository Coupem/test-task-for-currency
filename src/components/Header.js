import React from 'react';
import { HeaderComponent } from '../styled';

const Header = () => (
  <HeaderComponent>Best converter & valid currency course</HeaderComponent>
);

Header.propTypes = {};

export default Header;
