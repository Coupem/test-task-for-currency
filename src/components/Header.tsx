// eslint-disable-next-line @typescript-eslint/no-use-before-define
import React from 'react';
import { HeaderComponent } from '../styled';

const Header = (): JSX.Element => (
  <HeaderComponent>Best converter & valid currency course</HeaderComponent>
);

Header.propTypes = {};

export default Header;
