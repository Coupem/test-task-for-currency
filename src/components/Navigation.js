import React from 'react';
import { useHistory, useLocation } from 'react-router-dom';

import { CONVERTER_URL, COURSE_URL } from '../constants';
import { NavButton, NavigationComponent } from '../styled';

const Navigation = () => {
  const history = useHistory();
  const { pathname } = useLocation();

  const onClickButtonConverter = () => {
    history.push('/');
  };

  const onClickButtonCourse = () => {
    history.push('/course');
  };

  return (
    <NavigationComponent>
      <NavButton
        primary={pathname === CONVERTER_URL}
        onClick={onClickButtonConverter}
      >
        Converter
      </NavButton>
      <NavButton
        primary={pathname === COURSE_URL}
        onClick={onClickButtonCourse}
      >
        Course
      </NavButton>
    </NavigationComponent>
  );
};

Navigation.propTypes = {};

export default Navigation;
