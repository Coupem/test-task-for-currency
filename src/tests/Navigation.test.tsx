import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

import { Navigation } from '../components/index';
import { NavButton } from '../styled';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: 'localhost:3000/',
  }),
}));

describe('Navigation component', () => {
  const wrapper = shallow(<Navigation />);

  it('should render correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Test button component', () => {
    const mockCallBack = jest.fn();
    const button = shallow(<NavButton onClick={mockCallBack} />);
    button.find('button').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});
