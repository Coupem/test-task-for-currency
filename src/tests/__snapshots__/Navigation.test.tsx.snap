// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component should render correctly 1`] = `
<Styled(Col)>
  <styled.button
    onClick={[Function]}
    primary={false}
  >
    Converter
  </styled.button>
  <styled.button
    onClick={[Function]}
    primary={false}
  >
    Course
  </styled.button>
</Styled(Col)>
`;
